apps/devhub/tasks.py

```diff
@@ -272,17 +281,44 @@ def run_validator(path, for_appversions=None, test_all_tiers=False,
             path = temp.name
 
         with statsd.timer('devhub.validator'):
-            return validate(path,
-                            for_appversions=for_appversions,
-                            format='json',
-                            # When False, this flag says to stop testing after
-                            # one tier fails.
-                            determined=test_all_tiers,
-                            approved_applications=apps,
-                            spidermonkey=settings.SPIDERMONKEY,
-                            overrides=overrides,
-                            compat_test=compat,
-                            listed=listed)
+            json_result = validate(
+                path,
+                for_appversions=for_appversions,
+                format='json',
+                # When False, this flag says to stop testing after one
+                # tier fails.
+                determined=test_all_tiers,
+                approved_applications=apps,
+                spidermonkey=settings.SPIDERMONKEY,
+                overrides=overrides,
+                compat_test=compat,
+                listed=listed
+            )
+
+        track_validation_stats(json_result)
+
+        return json_result
+
+
+def track_validation_stats(json_result):
+    """
+    Given a raw JSON string of validator results, log some stats.
+    """
+    result = json.loads(json_result)
+    result_kind = 'success' if result['errors'] == 0 else 'failure'
+    statsd.incr('devhub.validator.results.all.{}'.format(result_kind))
+
+    if result['metadata']['listed']:
+        statsd.incr('devhub.validator.results.listed.{}'
+                    .format(result_kind))
+    else:
+        statsd.incr('devhub.validator.results.unlisted.{}'
+                    .format(result_kind))
+
+    if addon_can_be_signed(result):
```

Comment:
hmm, I'm not sure. I'm assuming an unlisted result is one that was uploaded for the first time, validated, but not yet approved (or uploaded just as a test). I think they might want to know how many of those could be signed automatically, no? It would be easy to capture data on listed and unlisted signing results.